lib/wallet/walletdb.js

```diff
@@ -1622,8 +1624,14 @@ class WalletDB extends EventEmitter {
         if (!map)
           continue;
 
-        for (const wid of map.wids)
-          wids.add(wid);
+        for (const wid of map.wids) {
+          if (await this.has(wid)) {
+            wids.add(wid);
+          } else {
+            await this.removeOutpointMap(b, hash, index, wid);
```

Comment:
I think `getWalletsByTX()` is the wrong place to be deleting records from the database. 
If you look at `_remove()`: https://github.com/handshake-org/hsd/blob/c85d9b4c743a9e1c9577d840e1bd20dee33473d3/lib/wallet/walletdb.js#L995-L1094

I see `removePathMap()`, `removeBlockMap()`, `removeOutpointMap()`, and `removeTXMap()` along with the other per-wallet buckets being wiped. 

So I think you are right that `_remove()` might be missing `removeNameMap()` and I think that should be the first thing to check if you are seeing leftover data causing unexpected behavior after removing a wallet.

The function `getWalletsByTX()` should not do anything except tell you which wallets are interested in a transaction :-) adding extra database calls here I think is the wrong approach and will confuse future developers.

In the same vein, the test should be able check that all data is cleared from the database after just calling `remove()`